static/js/message_list.js

```diff
@@ -251,21 +255,20 @@ class MessageList {
         let trailing_bookend_content;
         let show_button = true;
         const subscribed = stream_data.is_subscribed(stream_name);
-        if (subscribed) {
+        const sub = stream_data.get_sub(stream_name);
+        if (sub === undefined) {
+            trailing_bookend_content = this.deactivated_bookend_content();
+            // Hide the resubscribe button for streams that no longer exist.
+            show_button = false;
+        } else if (subscribed) {
             trailing_bookend_content = this.subscribed_bookend_content(stream_name);
         } else {
             if (!this.last_message_historical) {
                 trailing_bookend_content = this.unsubscribed_bookend_content(stream_name);
 
-                // For invite only streams or streams that no longer
-                // exist, hide the resubscribe button
+                // For invite only streams hide the resubscribe button
                 // Hide button for guest users
-                const sub = stream_data.get_sub(stream_name);
-                if (sub !== undefined) {
-                    show_button = !page_params.is_guest && !sub.invite_only;
-                } else {
-                    show_button = false;
-                }
+                show_button = !page_params.is_guest && !sub.invite_only;
```

Comment:
@clarammdantas I guess we should make sure your PR changes this line when we adjust the guest policies.